### PR TITLE
Upgrade the dev version of Airflow to 2.7.0

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
We already have a few PRs that have wrongly generated version (https://github.com/apache/airflow/pull/30827 for example) so likely it's a good time to merge 2.6.0.dev0 -> 2.7.0.dev0 :)

**Comment by potiuk** [comment]
Not sure which part of the release process it is in, but it looks like it shoudl be done slightly earlier (i.e. when we cut-off the 2.6.0 branch).

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -250,6 +250,13 @@ daemon_umask = 0o077
 # Example: dataset_manager_kwargs = {{"some_param": "some_value"}}
 # dataset_manager_kwargs =
 
+# (experimental) Whether components should use Airflow Internal API for DB connectivity.
```
Comment (pierrejeambrun) [inline airflow/config_templates/default_airflow.cfg:253]: Why do we have this change, this seems unrelated ?

Comment (potiuk) [reply airflow/config_templates/default_airflow.cfg:253]: Quite the contrary. They are very much related. They were (correctly) automatically added by pre-commit.

Those parameters have version-added = 2.7.0. This was implemented as part of the feature flags for AIP-44 and AIP-52, but in order to not comploicate it further  we restricted automated moving parameters from .yaml to config based on "version-added" > currrent version. We moved the related parameters that were scheduled for 2.7 to have "version-added=2.7.0" so they started to appear only when we moved the version to 2.7.0.

**Review by jedcunningham — APPROVED** [review]
